butterfree/migrations/database_migration/database_migration.py

```diff
@@ -0,0 +1,93 @@
+"""Migration entity."""
+
+from abc import ABC, abstractmethod
+from typing import Any, Dict, List, Optional
+
+from butterfree.transform import FeatureSet
+
+
+class DatabaseMigration(ABC):
+    """Abstract base class for Migrations."""
+
+    @abstractmethod
+    def create_query(
+        self,
+        table_name: str,
+        db_schema: List[Dict[str, Any]] = None,
+        diff_schema: List[Dict[str, Any]] = None,
+    ) -> Any:
+        """Create a query regarding a data source.
+
+        Returns:
+            The desired query for the given database.
+
+        """
+
+    def _apply_migration(self, feature_set: FeatureSet) -> None:
+        """Apply the migration in the respective database."""
+        pass
+
+    def _validate_schema(
+        self, fs_schema: List[Dict[str, Any]], db_schema: List[Dict[str, Any]] = None,
+    ) -> Any:
+        """Provides schema validation for feature sets.
+
+        Compares the schema of your local feature set to the
+        corresponding table in a given database.
+
+        Args:
+            fs_schema: object that contains feature set's schemas.
+            db_schema: object that contains the table og a given db schema.
+
+        """
+        mismatches = []
+
+        if not db_schema:
+            return fs_schema
+
+        for feature in fs_schema:
```

Comment:
Considering that both ```fs_schema``` and ```db_schema``` are lists of dictionaries, would it be possible to compare these two objects using ```difference``` or ```symmetric_difference```?